src/frontend/swo/sources/socket.ts

```diff
@@ -135,6 +137,39 @@ export class SocketRTTSource extends SocketSWOSource {
     }
 }
 
+export class DefmtSocketRTTSource extends SocketRTTSource {
+    private process: ChildProcess.ChildProcess;
+    private executable: string;
+    private cwd: string;
+
+    protected processData(buffer: Buffer): void {
+        this.process.stdin.write(buffer);
+    }
+
+    public start(timeout = (1000 * 60 * 5)): Promise<void> {
+        this.process = ChildProcess.spawn(
+            'defmt-print',['-e', this.executable], { cwd: this.cwd }
+        );
+
+        this.process.on('error', (e) => {
+            (e as any).message = `Failed to launch defmt-print (is it installed?)`; 
+            this.emit('error', e);
+            this.dispose();
+        });
+
+        this.process.stdout.on('data', (buffer) => this.emit('data', buffer));
+
+        return super.start(timeout);
+    }
+
+    constructor(tcpPort: string, public readonly channel: number, executable: string, wsPath: string) {
+        super(tcpPort, channel);
+
+        this.cwd = wsPath;
+        this.executable = Path.relative(wsPath, executable);
```

Comment:
Why?

Reply:
@haneefdm I checked the source of defmt and probe-rs. This is because by default, defmt's message is compressed. The detailed info is stored in .elf file, that's why the executable is needed: without .elf, the message cannot be parsed.

See https://github.com/knurling-rs/defmt/blob/main/decoder/src/lib.rs#L157 . The `Table` is what's stored in .elf file